embedded-graphics/src/primitives/triangle/mod.rs

```diff
@@ -232,12 +275,29 @@ mod tests {
 
     #[test]
     fn contains() {
-        let triangle = Triangle::new(Point::new(5, 10), Point::new(15, 20), Point::new(10, 15));
+        let triangles = [
+            // Colinear triangle. Mathematically, zero sized, in e-g land however this draws pixels
+            // along a line, therefore it is possible to contain a point.
```

Comment:
We might want to revisit colinear triangles later. To be consistent with rectangles, filling a zero sized triangles shouldn't draw any pixels. Can you create a new issue or add this to another issue so we remember this?

Reply:
I've implemented that in this PR. `contains()` now checks for colinearity by comparing the determinant with zero, as does `ScanlineIterator` which won't return any points for a colinear triangle either.

Reply:
`ScanlineIterator` should still draw a stroke for a colinear triangle, to be consistent with the other primitives.

Reply:
Oops, you're right. Fixed!